core/internal/cltest/cltest.go

```diff
@@ -649,7 +641,7 @@ func NewStoreWithConfig(t testing.TB, c config.GeneralConfig, flagsAndDeps ...in
 	t.Cleanup(func() {
 		cleanUpStore(t, s)
 	})
-	s.Config.SetDB(s.DB)
+	c.SetDB(s.DB)
```

Comment:
I never really liked this SetDB thing, I think the db should be passed into the config instead

Reply:
Yeah I'd also like to do this, it'll be easier to solve after this & https://github.com/smartcontractkit/chainlink/pull/5013 are merged. Initializing a connection currently reads a bunch of config variables.